engine/classes/Elgg/Di/ServiceProvider.php

```diff
@@ -242,6 +244,10 @@ public function __construct(\Elgg\AutoloadManager $autoload_manager) {
 		});
 
 		$this->setClassName('widgets', '\Elgg\WidgetsService');
+		
+		$this->setFactory('uploads', function(ServiceProvider $c) {
+			return $c->request->files;
```

Comment:
Why do we need a separate entry for this? If we moved some uploads-related functions into a new UploadsService we could return that here.